libr/bin/format/xtac/bin_xtac.c

```diff
@@ -0,0 +1,587 @@
+
+		if (r_buf_read_at (bin->b, p_buffer, (ut8*) entry, sizeof (ut32)) < 0) {
+			R_LOG_WARN ("Warning: read (xtac linked list metadata)\n");
+			free (entry);
+			return false;
+		}
+
+		const ut32 meta = GET_META_DATA(entry->meta_and_offset);
+		if (has_forward_edge_addr (meta)) {
+			p_buffer += sizeof (ut32);
+			if (r_buf_read_at (bin->b, p_buffer, (ut8*) &entry->forward_edge_addr, sizeof (ut32)) < 0) {
+				R_LOG_WARN ("Warning: read (xtac linked list forward edge address)\n");
+				free (entry);
+				return false;
+			}
+		}
+		if (has_backward_edge_addr (meta)) {
+			p_buffer += sizeof (ut32);
+			if (r_buf_read_at (bin->b, p_buffer, (ut8*) &entry->backward_edge_addr, sizeof (ut32)) < 0) {
+				R_LOG_WARN ("Warning: read (xtac linked list backward edge address)\n");
+				free (entry);
+				return false;
+			}
+		}
+
+		r_list_append(bin->xtac_linked_list, entry);
+		p_xtac_linked_list_entry += GET_OFFSET(entry->meta_and_offset) * 4;
+	} while (p_xtac_linked_list_entry < bin->header->ptr_to_addr_pairs);
+
+	if (GET_OFFSET(entry->meta_and_offset) != 0x0FFFFFFF) {
+		R_LOG_WARN ("Warning: xtac linked list is not properly terminated\n");
+		R_LOG_WARN ("Some entry of xtac linked list might be broken\n");
+	}
+
+	return true;
+}
+
+static bool r_bin_xtac_init(RBinXtacObj *bin) {
+	// NOTE: not tested on big-endian processor
+	if (!r_bin_xtac_read_header(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_address_pairs(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_module_name(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_nt_native_pathname(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_blck_stubs(bin)) {
+		return false;
+	}
+
+	if (!r_bin_xtac_read_xtac_linked_list(bin)) {
+		return false;
+	}
+
+	return true;
+}
+
+static void* r_bin_xtac_free(RBinXtacObj *bin) {
+	if (!bin) {
+		return NULL;
+	}
+	free (bin->header);
+	free (bin->address_pairs);
+	free (bin->mod_name_u16);
+	free (bin->mod_name_u8);
+	free (bin->nt_path_name_u16);
+	free (bin->nt_path_name_u8);
+
+	r_buf_free (bin->b);
+	bin->b = NULL;
+
+	r_list_free (bin->blck_stubs);
+	r_list_free (bin->xtac_linked_list);
+	bin->blck_stubs = NULL;
+	bin->xtac_linked_list = NULL;
+
+	free (bin);
+	return NULL;
+}
+
+static RBinXtacObj *r_bin_xtac_new_buf(RBuffer *buf, bool verbose) {
+	RBinXtacObj *bin = R_NEW0 (RBinXtacObj);
+	if (!bin) {
+		return NULL;
+	}
+	bin->b = r_buf_ref (buf);
+	bin->size = r_buf_size (buf);
+	bin->verbose = verbose;
+	if (!r_bin_xtac_init (bin)) {
+		return r_bin_xtac_free(bin);
+	}
+	return bin;
+}
+
+static bool load_buffer(RBinFile* bf, void **bin_obj, RBuffer *buf, ut64 loadaddr, Sdb *sdb) {
+	r_return_val_if_fail (bf && bin_obj && buf, false);
+	RBinXtacObj *res = r_bin_xtac_new_buf(buf, bf->rbin->verbose);
+	if (res) {
+		*bin_obj = res;
+		return true;
+	}
+	return false;
+}
+
+static void destroy(RBinFile *bf) {
+	r_bin_xtac_free ((RBinXtacObj*)bf->o->bin_obj);
+}
+
+static ut64 baddr(RBinFile* bf) {
```

Comment:
```suggestion
static ut64 baddr(R_UNUSED RBinFile* bf) {
```

this macro works quite well in all modern compilers, no need to do the (void)bf, below